src/commands/generate/fromTemplate.ts

```diff
@@ -122,6 +132,9 @@ export default class Template extends Command {
       noOverwriteGlobs: flags['no-overwrite'],
       mapBaseUrlToFolder: parsedFlags.mapBaseUrlToFolder,
       disabledHooks: parsedFlags.disableHooks,
+      registryUrl: parsedFlags.registryUrl,
+      registryAuth: flags['registry-auth'],
```

Comment:
registry auth should be change with base 64 value as mentioned in  [this](https://www.asyncapi.com/docs/tools/generator/using-private-template)
You can do this in something like this
```bash
const registryAuth = registryUsername && registryPassword
    ? Buffer.from(`${registryUsername}:${registryPassword}`).toString('base64')
    : undefined;
```